.github/workflows/selfhosted.yml

```diff
@@ -0,0 +1,120 @@
+# CI workflow that runs on self hosted cluster
+name: Self-hosted rules
+
+on:
+  workflow_call:
+    inputs:
+      used-labels:
+        description: 'Label names in JSON format'
+        required: true
+        type: string
+      python-version:
+        description: 'Used python version'
+        required: true
+        type: string
+      slurm-job:
+        description: 'Name of the slurm job'
+        required: true
+        type: string
+      server-url:
+        description: ''
+        required: true
+        type: string
+      repository:
+        description: ''
+        required: true
+        type: string
```

Comment:
Since they only appear together, you can consider merging the two of them in a single parameter